addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -759,6 +771,28 @@ export class PosStore extends Reactive {
             this.hasJustAddedProduct = false;
         }, 3000);
 
+        // orderlines will be sorted on the basis of pos product category and sequence for group the orderlines in order cart
+        if (this.config.orderlines_sequence_in_cart_by_category && order.lines.length) {
+            order.lines.sort(this.sortBySequenceAndCategory);
+
+            const resultLines = [];
+
+            order.lines.forEach((line) => {
+                if (line.combo_line_ids?.length > 0) {
+                    resultLines.push(line);
+                    const childLines = order.lines.filter(
+                        (childLine) => childLine.combo_parent_id?.id === line.id
+                    );
+                    childLines
+                        .sort(this.sortBySequenceAndCategory)
+                        .forEach((childLine) => resultLines.push(childLine));
+                } else if (!line.combo_parent_id) {
+                    resultLines.push(line);
+                }
+            });
+            order.lines = resultLines;
+        }
+
```

Comment:
Don't mutate the lines. create a getter in the pos_order.js class that returns the sorted items. e.g.

```js
getSortedOrderlines() {
    if (tobesorted) { // <--- I think you have access with config here via `this.config_id`.
        return new array containing the sorted items.
    } else {
        return this.lines;
    }
}

Reply:
One more thing, don't forget the reward lines added by pos_loyalty.